crates/cli/tests/cli_run.rs

```diff
@@ -77,899 +48,224 @@ mod cli_run {
-                stdin,
-                &[LINKER_FLAG, "legacy"],
-                &roc_app_args,
-                extra_env,
-                expected_ending,
-                use_valgrind,
-                test_cli_commands,
-            );
-        }
-    }
-
-    #[test]
-    #[serial(zig_platform_parser_package_basic_cli_url)]
-    #[cfg_attr(windows, ignore)]
-    fn hello_world() {
-        test_roc_app_slim(
-            "examples",
-            "helloWorld.roc",
-            "Hello, World!\n",
-            UseValgrind::Yes,
-        )
-    }
-
-    #[cfg(windows)]
-    const LINE_ENDING: &str = "\r\n";
-    #[cfg(not(windows))]
-    const LINE_ENDING: &str = "\n";
-
-    #[test]
-    #[cfg_attr(windows, ignore)]
-    // uses C platform
-    fn platform_switching_main() {
-        test_roc_app_slim(
-            "examples/platform-switching",
-            "main.roc",
-            &("Which platform am I running on now?".to_string() + LINE_ENDING),
-            UseValgrind::Yes,
-        )
-    }
+    // fn check_compile_error(file: &Path, flags: &[&str], expected: &str) {
+    //     check_compile_error_with(CMD_CHECK, file, flags, expected);
+    // }
+
+    // fn check_compile_error_with(cmd: &str, file: &Path, flags: &[&str], expected: &str) {
+    //     let compile_out = run_roc([cmd, file.to_str().unwrap()].iter().chain(flags), &[], &[]);
+    //     let err = compile_out.stdout.trim();
+    //     let err = strip_colors(err);
+    //     #[cfg_attr(windows, ignore)]
+    //     // uses C platform
+    //     fn platform_switching_main() {
+    //         let expected_ending = "Which platform am I running on now?\n🔨 Building host ...\n";
+    //         let runner = Run::new_roc()
+    //             .arg(CMD_RUN)
+    //             .arg(BUILD_HOST_FLAG)
+    //             .arg(SUPPRESS_BUILD_HOST_WARNING_FLAG)
+    //             .add_arg_if(LINKER_FLAG, TEST_LEGACY_LINKER)
+    //             .with_valgrind(ALLOW_VALGRIND)
+    //             .arg(file_from_root("examples/platform-switching", "main.roc").as_path());
+
+    //         let out = runner.run();
+    //         out.assert_clean_success();
+    //         out.assert_stdout_and_stderr_ends_with(expected_ending);
+    //     }
 
     // We exclude the C platforming switching example
     // because the main platform switching example runs the c platform.
-    // If we don't, a race condition leads to test flakiness.
+    // If we don't, a race condition leads to test flakiness if we attempt
+    // to build the host from two different tests running concurrently.
 
     #[test]
     #[cfg_attr(windows, ignore)]
     fn platform_switching_rust() {
-        test_roc_app_slim(
-            "examples/platform-switching",
-            "rocLovesRust.roc",
-            "Roc <3 Rust!\n",
-            UseValgrind::Yes,
-        )
+        let expected_ending = "Roc <3 Rust!\n🔨 Building host ...\n";
+        let runner = Run::new_roc()
+            .arg(CMD_RUN)
```

Comment:
Assuming that `run` still tries to run a program even if there are errors, that is probably not what we want to use here...